src/matlab2tikz.m

```diff
@@ -5610,9 +5610,9 @@ function warnFacetedInterp(m2t)
     end
 end
 % ==============================================================================
-function [val, isDefault] = getAndCheckDefault(type, handle, key, default)
+function [val, isDefault] = getAndCheckDefault(handle, key, default)
     % gets the value from a handle of certain type and check the default values
-    default   = getFactoryOrDefault(type, key, default);
+    default   = getFactoryOrDefault(handle.Type, key, default);
```

Comment:
This fails in octave and MATLAB HG1. You might get this working using `get(handle, 'Type')` (untested).